data-pipeline/src/pipeline/database.py

```diff
@@ -805,3 +805,61 @@ def insert_la_non_financial(
         run_id=run_id,
         engine=engine,
     )
+
+
+def insert_la_statistical_neighbours(
+    run_type: str,
+    run_id: str,
+    df: pd.DataFrame,
+    engine: sqlalchemy.engine.Engine | None = None,
+):
+    """
+    Persist Local Authority Statistical Neighbour data to the database.
+
+    Note: the data must be un-pivoted to match the database schema.
+
+    :param run_type: "default" or "custom"
+    :param run_id: unique identifier for processing
+    :param df: Local Authority Statistical Neighbour information
+    :param engine: (optional) SQLAlchemy Engine
+    """
+    write_frame = df.reset_index().rename(
+        columns=config.la_statistical_neighbours_projections
+    )[["LaCode"] + [c for c in df.columns if c.startswith("SN")]]
+
+    _desc = write_frame.melt(
```

Comment:
I think I've talked myself around: I'll refactor this out and wrap it in some tests.